java/arcs/core/entity/EntityDereferencerFactory.kt

```diff
@@ -32,16 +33,21 @@ class EntityDereferencerFactory(
         if (value == null) return
         when (value) {
             is Reference -> value.dereferencer = create(schema)
-            is RawEntity -> injectDereferencers(schema, value)
+            is RawEntity -> injectDereferencersIntoRawEntity(schema, value)
             is Set<*> -> value.forEach { injectDereferencers(schema, it) }
+            is ReferencableList<*> -> value.value.forEach { injectDereferencers(schema, it) }
         }
     }
 
-    private fun injectDereferencers(schema: Schema, rawEntity: RawEntity) {
+    private fun injectDereferencersIntoRawEntity(schema: Schema, rawEntity: RawEntity) {
         fun injectField(fieldType: FieldType?, fieldValue: Any?) {
             val schemaHash = when (fieldType) {
                 is FieldType.EntityRef -> fieldType.schemaHash
                 is FieldType.InlineEntity -> fieldType.schemaHash
+                is FieldType.ListOf -> {
+                    injectField(fieldType.primitiveType, fieldValue)
+                    null
```

Comment:
injectField returns null, can you omit this line?

Reply:
I worry that it makes the code flow significantly less clear if I do. This is kind of documentation more than anything. WDYT?

Reply:
sounds good